File: week-1/Day_1-2/parseCollection.py
# Use json module to aide proper data structure of imported data.
import json
# Capture JSON data fropm file and store for use
with open("system_logs_day-1.json", "r") as file:
    data = json.load(file)
# Define iteration variable
i = 0
# Define tracking variable using event ID. Using Event ID saves on storage, but might not be as ideal on time.
events = []

# Check that data access is correct
# print(data[i]["events"][1]["payload"]["rows"])

while True:
    if i >= len(data):
        break
    else:
        if (data[i]["events"][0]["payload"]["status"].lower() == "fail" or data[i]["events"][1]["payload"]["rows"] > 100):
            events.append(data[i]["id"])
            # Debugging check
            # print("We have added an event")
        # Debugging check
        # print("i = " + str(i))
        i+=1

print("The following Event IDs are failing or have excessive row counts: " + str(events))

File: week-1/Day_1-2/parseCollection.js
const data = require("./system_logs_day-1.json");
//Check data
//console.log(data[1].events);

// We want to capture every event where status == "fail" OR rows > 100
// User .reduce() and .filter() to accomplish this.

const events = data.filter(checkIssue).reduce(extractID,[]);

/*
---------- Helper Functions ----------
*/

// Function to create test for capturing the appropriate log entries
function checkIssue(logEntry){
    let isIssue = false;

    for(let i = 0; i < logEntry.events.length; i++){
        if(logEntry.events[i].payload.status === "fail" || logEntry.events[i].payload.rows > 100){
            isIssue = true;
        }
    }

    return isIssue;
}

// Function to reduce the error logs down to just their IDs.
function extractID(accumulator, logEntry){
    accumulator.push(logEntry.id);
    return accumulator;
}

// Check out events variable to ensure the proper events are captured.
// console.log(events);